src/workerd/api/r2-rpc.c++

```diff
@@ -217,9 +215,15 @@ kj::Promise<R2Result> doR2HTTPPutRequest(jsg::Lock& js, kj::Own<kj::HttpClient>
             return reqBody->write(data.begin(), data.size());
           }
           KJ_CASE_ONEOF(stream, jsg::Ref<ReadableStream>) {
-            auto dest = newSystemStream(kj::mv(reqBody), StreamEncoding::IDENTITY, context);
-            return context.waitForDeferredProxy(streamSource->pumpTo(*dest, true))
-                .attach(kj::mv(streamSource), kj::mv(dest));
+            // Because the ReadableStream might be a fully JavaScript-backed stream, to
+            // pump to must be run within the IoContext so that we can run the pump inside
+            // the isolate lock.
```

Comment:
But only the initial `pumpTo()` call, right, not subsequent I/O operations related to the `pumpTo()`? I ask, because subjecting internal streams' individual I/O operations to the request's CPU time limit could be a potentially breaking change.

Reply:
Yes, i'll clarify. I see there are a few other typos here too.